crates/cxx-qt-gen/src/naming/name.rs

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use quote::format_ident;
+use syn::{spanned::Spanned, Attribute, Ident, Result};
+
+use crate::syntax::{attribute::attribute_find_path, expr::expr_to_string};
+
+/// This struct contains all names a certain syntax element may have
+///
+/// This includes the rust_name, cxx_name, as well as qualifications like
+/// the rust module and C++ namespace.
+#[derive(Debug)]
+pub struct Name {
+    pub(super) rust: Ident,
+    pub(super) cxx: Option<String>,
```

Comment:
does this refer to the C++ name or the CXX bridge name ? (it's the C++ name, but not initially obvious :thinking: )

But i guess `.cxx` is better than `.cpp` as it's come from `cxx_name`. Maybe just add a doc line onto these fields to say where they come from. Eg from the `cxx_name` attribute or the `namespace` attribute etc.

Reply:
Yeah, the idea is that access would be something like: `qenum.name.cxx`.
Which should be rather obvious that that's the `cxx_name`.

Reply:
Added additional doc comments